### PR TITLE
Rename Min, Max operations to MinOf, MaxOf, and If effect to Conditional in python parser

### PR DESCRIPTION
**Pull request by o01eg**
This change allows to not conflict with statistic type enums.

### CODE REVIEW

File: parse/ValueRefPythonParser.cpp [hunk 1]
```diff
@@ -350,10 +349,17 @@ void RegisterGlobalsValueRefs(boost::python::dict& globals, const PythonParser&
...
-    std::function<boost::python::object(const boost::python::tuple&, const boost::python::dict&)> f_insert_min = [&parser](const boost::python::tuple& args, const boost::python::dict& kw) { return insert_minmaxoneof_<ValueRef::OpType::MINIMUM>(parser, args, kw); };
-    globals["Min"] = boost::python::raw_function(f_insert_min, 3);
-    std::function<boost::python::object(const boost::python::tuple&, const boost::python::dict&)> f_insert_max = [&parser](const boost::python::tuple& args, const boost::python::dict& kw) { return insert_minmaxoneof_<ValueRef::OpType::MAXIMUM>(parser, args, kw); };
-    globals["Max"] = boost::python::raw_function(f_insert_max, 3);
+
+    // selection_operator
+    for (const auto& op : std::initializer_list<std::pair<const char*, ValueRef::OpType>>{
+            {"OneOf",   ValueRef::OpType::RANDOM_PICK},
+            {"MinOf",   ValueRef::OpType::MINIMUM},
+            {"MaxOf",   ValueRef::OpType::MAXIMUM}})
```
Comment (geoffthemedio) [inline parse/ValueRefPythonParser.cpp:357]: not sure if any better / different, but could also do:
```suggestion
    for (const auto& op : std::array<std::pair<const char*, ValueRef::OpType>, 3> {{
            {"OneOf",   ValueRef::OpType::RANDOM_PICK},
            {"MinOf",   ValueRef::OpType::MINIMUM},
            {"MaxOf",   ValueRef::OpType::MAXIMUM}}})
```

File: default/scripting/techs/construction/FORCE_ENERGY_STRC.focs.py [hunk 2]
```diff
@@ -16,68 +16,68 @@
             accountinglabel="CON_TECH_ACCOUNTING_LABEL",
             priority=AFTER_ALL_TARGET_MAX_METERS_PRIORITY,
             effects=[
-                If(
+                EffectIf(
```
Comment (geoffthemedio) [inline default/scripting/techs/construction/FORCE_ENERGY_STRC.focs.py]: this seems awkward

Comment (Oberlus) [reply default/scripting/techs/construction/FORCE_ENERGY_STRC.focs.py]: `ConditionIf` ?

Comment (geoffthemedio) [reply default/scripting/techs/construction/FORCE_ENERGY_STRC.focs.py]: internally it's called `Conditional` ... Can't think of anything better than that.

File: default/scripting/techs/construction/FORCE_ENERGY_STRC.focs.py [hunk 3]
```diff
@@ -16,68 +16,68 @@
...
                     condition=(Value(LocalCandidate.Industry) <= Value(LocalCandidate.TargetIndustry)),
```
Comment (geoffthemedio) [inline default/scripting/techs/construction/FORCE_ENERGY_STRC.focs.py:20]: could change the `condition=` to `if=` for consistency with the `else=` below perhaps?

File: default/scripting/techs/construction/FORCE_ENERGY_STRC.focs.py [hunk 4]
```diff
@@ -16,68 +16,68 @@
...
                     effects=SetIndustry(
```
Comment (geoffthemedio) [inline default/scripting/techs/construction/FORCE_ENERGY_STRC.focs.py:21]: perhaps make `effects=` be `matching_effects=` and `else=` be `non_matching_effects=` ?

Comment (o01eg) [reply default/scripting/techs/construction/FORCE_ENERGY_STRC.focs.py:21]: I'd like to change original FOCS tokens and structure as little as possible.

File: parse/ValueRefPythonParser.cpp (hunk 1 above)
Comment (o01eg) [reply parse/ValueRefPythonParser.cpp:357]: I use same loop in other places so not sure if forcing to define size is better.

File: default/scripting/techs/construction/FORCE_ENERGY_STRC.focs.py (hunk 2 above)
Comment (o01eg) [reply default/scripting/techs/construction/FORCE_ENERGY_STRC.focs.py]: Changed to it.

**Comment by Cjkjvfnby** [comment]
> This change allows to not conflict with statistic type enums.

Should we just rename statistic type enums?  I guess conditions will e used more often so they are deserve shorter names.

**Comment by o01eg** [comment]
Not sure what if better, statistic type enums are expected to use with token `Statistic` which already long.

**Pull request merged**